### ISSUE TITLE
Consider changing CI systems

### ISSUE THREAD

**Issue by dhui** [golang-migrate/migrate#252]
## Why?
TravisCI builds are slow and flaky. Other open source projects have migrated from TravisCI to CircleCI.

### Must have
* Runs tests in multiple versions of Go
* Runs Docker daemon (for db tests)
    * Min Docker daemon version: `18.09`. See: https://github.com/golang-migrate/migrate/issues/254
* Integrations
    * Code coverage (coveralls or codecov)
    * Deploy 
        * Github assets
        * DockerHub
        * packagecloud
### Nice to have
* Fast
    * Builds triggered soon after pushes/merges
    * Shorter build times
        * Access to faster compute or I/O
        * Runs tests in parallel

## CI Solutions to Consider:
* CircleCI
* ~BuildKite~ - Need to host own workers/agents
* GitHub Actions

**Comment by thaJeztah** [comment]
Opened https://github.com/golang-migrate/migrate/pull/270 to fix the Docker API version mismatch

**Comment by dhui** [comment]
TravisCI is [no longer supporting open source projects](https://news.ycombinator.com/item?id=25338983), which blocks merges and releases for us, so this issue is higher priority.

**Comment by dhui** [comment]
Update:
* I got tests running in CircleCI but don't have coverage reporting yet. Once coverage reporting is done, then merges will be unblocked.
* Releases will be blocked until we can get assets/artifacts to be produced by CircleCI

**Comment by dhui** [comment]
Builds with coverage reports are now working on CircleCI, so merges are unblocked.

However, releases are still blocked.

**Comment by johejo** [comment]
How about goreleaser?
https://github.com/goreleaser/goreleaser
I will try to create a PR.

**Comment by dhui** [comment]
@johejo Using goreleaser would unblock releases. Any help is appreciated! See also: #74

This may also fix @Juneezee's problem of [supporting multiple arch](https://goreleaser.com/customization/docker/) by enabling `buildx`!

**Comment by dhui** [comment]
Switched to use Github Actions with GoReleaser and releases are now unblocked. See [v4.15.0-beta.1](https://github.com/golang-migrate/migrate/releases/tag/v4.15.0-beta.1)
